prometheus/src/main/java/org/opensearch/sql/prometheus/client/PrometheusClientImpl.java

```diff
@@ -78,9 +81,12 @@ public Map<String, List<MetricMetadata>> getAllMetrics() throws IOException {
         .build();
     Response response = this.okHttpClient.newCall(request).execute();
     JSONObject jsonObject = readResponse(response);
-    TypeReference<HashMap<String, List<MetricMetadata>>> typeRef
-        = new TypeReference<>() {};
-    return new ObjectMapper().readValue(jsonObject.getJSONObject("data").toString(), typeRef);
+    Type typeTok = new TypeToken<HashMap<String, List<MetricMetadata>>>() {}.getType();
+    Gson gson = new GsonBuilder()
```

Comment:
Could the creation of the Gson builder be refactored into common or core so it can be retrieved by all parts of this project, reduces copypasta based issues.

Reply:
Do you mean create a static variable with the minimum reflection requirements?

Reply:
From https://stackoverflow.com/questions/35409844/can-we-reuse-a-gson-object
> The Gson instance does not maintain any state while invoking Json operations. So, you are free to reuse the same object for multiple Json serialization and deserialization operations.

Seems like a static instance would be fine, but I would recommend exposing it via a function. e.g. `public Gson getInstance() {return GSON_INSTANCE;}` so it can be mocked if needed in tests systems.